docs/csharp/language-reference/keywords/is.md

```diff
@@ -31,35 +32,135 @@ translation.priority.ht:
+
+[!code-cs[is#2](../../../../samples/snippets/csharp/language-reference/keywords/is/is2.cs)]
+
+`expr` can be any expression that returns a value, with the exception of anonymous methods and lambda expressions. The following example uses  `is` to evaluate the return value of a method call.   
+[!code-cs[is#4](../../../../samples/snippets/csharp/language-reference/keywords/is/is4.cs)]
+
+Starting with C# 7, you can use pattern matching with the [type pattern](#type) to write more concise code that uses the `is` statement.
+
+## Pattern matching with `is` ##
+
+Starting with C# 7, the `is` and [switch](../../../csharp/language-reference/keywords/switch.md) statements support pattern matching. The `is` keyword supports the following patterns:
+
+- [Type pattern](#type),  which tests whether an expression can be converted to a specified type and, if it can be, casts it to a variable of that type.
+
+- [Constant pattern](#constant), which tests whether an expression evaluates to a specified constant value.
+
+- [var pattern](#var), a match that always succeeds and binds the value of an expression to a new local variable. 
+
+
+### <a name="type" /> Type pattern </a>
+
+When using the type pattern to perform pattern matching, `is` tests whether an expression can be converted to a specified type and, if it can be, casts it to a variable of that type. It is a straightforward extension of the `is` statement that enables concise type evaluation and conversion. The general form of the `is` type pattern is:
+
+```cs
+   expr is type varname 
+```
+
+where *expr* is an expression that evaluates to an instance of some type, *type* is the name of the type to which the result of *expr* is to be converted, and *varname* is the object to which the result of *expr* is converted if the `is` test is `true`. 
+
+The `is` expression is `true` if any of the following is true:
+
+- *expr* is an instance of the same type as *type*.
+
+- *expr* is an instance of a type that derives from *type*. In other words, the result of *expr* can be upcast to an instance of *type*.
+
+- *expr* is an instance of a type that is a base class of *type*, and *expr* is the result of a downcast from *type* or one of its derived types.
+
+- *expr* is an instance of a type that implements the *type* interface.
+
+If *exp* is `true` and `is` is used with an `if` statement, *varname* is assigned and has local scope within the `if` statement only.
+
+The following example uses the `is` type pattern to provide the implementation of a type's <xref:System.IComparable.CompareTo(System.Object)?displayProperty=fullName> method.
+
+[!code-cs[is#5](../../../../samples/snippets/csharp/language-reference/keywords/is/is-type-pattern5.cs)]
+
+Without pattern matching, this code might be written as follows. The use of type pattern matching produces more compact, readable code by eliminating the need to test whether the result of a conversion is a `null`.  
```

Comment:
I'd consider adding an example using is with a value type, because the previous version of that could would require `as` and an explicit cast, or a cast inside a `try` / `catch` block.